RHOAI-16971 LAB-tuning - user (#775)
opendatahub-io/opendatahub-documentation: modules/using-lab-tuning.adoc

+
The *Model registry* page opens.
. Click the name of the model you registered.
+
The details page for the registered model opens.
. On the *Versions* tab of the model details page, click the name of the version you want to LAB-tune.
+
The details page for the version opens.
. Review the version details.
. Click *LAB-tune*.
+
The *Start a LAB-tuning run* dialog opens.
. Select your project from the *Data science project* drop-down list.
. Click *Continue to run details*.
+
. The *Start a LAB-tuning run* page opens.
. Review the *Project details*, *Pipeline details*, and *Base model* sections for accuracy.
. In the *Taxonomy details* section, enter your *Taxonomy GIT URL*. 
. If your Git repository requires authentication, select either the *SSH key* or *Username and token* method and enter the appropriate credentials.
. In the *LAB teacher model* section, if your teacher model requires token authentication, select *Authenticated endpoint* and enter the *Endpoint*, *Token*, and *Model name*. If your teacher model does not require token authentication, select *Unauthenticated endpoint* and enter the *Endpoint* and *Model name*. 
+
[TIP]
====

> To find token authentication for your judge and teacher models, go to the *Models* tab of your data science project. For *Endpoint* and *Model name*, click *Internal and external endpoint details* for your model. For *Token*, expand the section for your model and find the *Token authentication* section.

Reviewer: Important: for both models, users need to add `/v1` at the end of the url of the endpoint provided by RHOAI when configuring the LAB-tuning process

Example:  https://mixtral-my-project.apps.my-cluster.com/v1

* You have access to an available model registry in your deployment.
* You have created your taxonomy, prepared an OCI storage location for the LAB-tuned model, created a data science project, and deployed teacher and judge models, as described in _Preparing LAB-tuning resources_.
* You have registered a base model with the `LAB starter` label from the model catalog, as described in _Registering a base model_.

.Procedure
. From the {productname-short} dashboard, click *Models* -> *Model registry*.
+
The *Model registry* page opens.
. Click the name of the model you registered.
+
The details page for the registered model opens.
. On the *Versions* tab of the model details page, click the name of the version you want to LAB-tune.
+
The details page for the version opens.
. Review the version details.
. Click the *LAB-tune* button.
+
The *Start a LAB-tuning run* dialog opens.
. Select your project from the *Data science project* drop-down list.
. Click *Continue to run details*.
+
. The *Start a LAB-tuning run* page opens.
. Review the *Project details*, *Pipeline details*, and *Base model* sections for accuracy.

> . In the *Taxonomy details* section, enter your *Taxonomy GIT URL*. 

Author: Right, that's how it is in the UI.

. If your Git repository requires authentication, select either the *SSH key* or *Username and token* method and enter the appropriate credentials.

> . In the *LAB teacher model* section, if your teacher model requires token authentication, select *Authenticated endpoint* and enter the *Endpoint*, *Token*, and *Model name*. If your teacher model does not require token authentication, select *Unauthenticated endpoint* and enter the *Endpoint* and *Model name*. 

Reviewer 2: I do see why it's in this order originally, and am fine if you want to leave it as-is - but think it adds to cognitive load to have it that way.

> . In the *LAB teacher model* section, if your teacher model requires token authentication, select *Authenticated endpoint* and enter the *Endpoint*, *Token*, and *Model name*. If your teacher model does not require token authentication, select *Unauthenticated endpoint* and enter the *Endpoint* and *Model name*. 

Author: I can improve this wording later.

====
. In the *Training hardware* section, configure the following settings:
.. For *Hardware profile*, select a hardware profile to match the hardware requirements of your workload to available node resources. 
.. For *Training nodes*, enter the total number of nodes to use for the run. Note that one node is used for the evaluation run phase.
.. For *Storage class*, select a storage class that is compatible with LAB-tuning and distributed training.
. Optional: In the *Hyperparameters* section, configure advanced settings for the run.
. In the *Fine-tuned model details* section, configure settings for the fine-tuned version of the base model:
.. For *Model output storage location*, select an *Existing connection* location to store the fine-tuned model output, or select *Create connection* to create a new connection.
.. For *OCI storage location* field, enter the full *Model URI* where the LAB-tuned model will be stored (for example, `oci://quay.io/my-org/fine-tuned-model-name:version`). Note that this value is separate from the connection. The connection provides access, while the URI defines the specific location.

> . Select the *Add model to registry* check box to enable storing, sharing, versioning, and deploying the LAB-tuned model in the model registry.

Reviewer 2: ```suggestion
. Select the *Add model to registry* check box so you can store, share, version, and deploy the LAB-tuned model in the model registry.
```

.. For *Model version name*, enter a name for the new LAB-tuned model version.
. Click *Start run*.

.Verification
* The *Runs* page opens for the pipeline version.
//Other ways to verify?

== Monitoring your LAB-tuning run

To monitor the status of your LAB-tuning run, follow these steps:

. From the {productname-short} dashboard, click *Data science pipelines* -> *Runs*. 
. Select your project from the *Project* list. 
. Check the status for the run. For more information, see _Viewing active pipeline runs_.
. When the status is `Succeeded`, click the name of the run to view the pipeline graph and details. For more information, see _Viewing the details of a pipeline version_.

== Reviewing and deploying your LAB-tuned model

> When the pipeline run is finished, your LAB-tuned model is available in the storage location you specified during the LAB-tune run configuration.

Reviewer 2: ```suggestion
When the pipeline run is finished, your LAB-tuned model is available in the storage location that you specified during the LAB-tune run configuration.
```

> If you selected the *Add model to registry* check box when you configured the run, the LAB-tuned model is in the model registry as a new version of the registered base model. 

Reviewer 2: ```suggestion
If you selected *Add model to registry* when you configured the run, the LAB-tuned model is in the model registry as a new version of the registered base model. 
```

To view and deploy your LAB-tuned model from the model registry, follow these steps:

. From the {productname-short} dashboard, click *Models* > *Model registry*.
+
The *Model registry* page opens.
. Click the name of the base model you registered.
+
The details page for the registered model opens.
. On the *Versions* tab of the model details page, click the name of the new version.
+
The details page for the version opens.

> . To deploy the LAB-tuned model, click the *Deploy* button.

Reviewer 2: ```suggestion
. To deploy the LAB-tuned model, click *Deploy*.
```
